maven-scm-providers/maven-scm-providers-git/maven-scm-provider-jgit/src/main/java/org/apache/maven/scm/provider/git/jgit/command/PlexusInteractivityCredentialsProvider.java

```diff
@@ -0,0 +1,106 @@
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+
+import java.util.Arrays;
+
+import org.codehaus.plexus.components.interactivity.Prompter;
+import org.codehaus.plexus.components.interactivity.PrompterException;
+import org.eclipse.jgit.errors.UnsupportedCredentialItem;
+import org.eclipse.jgit.transport.CredentialItem;
+import org.eclipse.jgit.transport.CredentialsProvider;
+import org.eclipse.jgit.transport.URIish;
+
+/**
+ * {@link CredentialsProvider} leveraging the {@link Prompter} component.
+ *
+ */
+public class PlexusInteractivityCredentialsProvider extends CredentialsProvider
+{
+    private boolean isInteractive;
+    private final Prompter prompter;
+
+    public PlexusInteractivityCredentialsProvider( Prompter prompter )
+    {
+        this.isInteractive = true;
+        this.prompter = prompter;
+    }
+
+    @Override
+    public boolean get( URIish uri, CredentialItem... items ) throws UnsupportedCredentialItem
+    {
+        for ( CredentialItem item : items )
+        {
+            try 
+            {
+                if ( item instanceof CredentialItem.YesNoType )
+                {
+                    CredentialItem.YesNoType yesNoItem = ( CredentialItem.YesNoType ) item;
+                    String value = prompter.prompt( item.getPromptText(), Arrays.asList( "yes", "no" ) );
+                    yesNoItem.setValue( value.equals( "yes" ) );
+                } 
+                else if ( item instanceof CredentialItem.Password )
+                {
+                    CredentialItem.Password passwordItem = ( CredentialItem.Password ) item;
+                    String password = prompter.promptForPassword( passwordItem.getPromptText() );
+                    passwordItem.setValue( password.toCharArray() );
+                }
+                else if ( item instanceof CredentialItem.Username )
+                {
+                    CredentialItem.Username usernameItem = ( CredentialItem.Username ) item;
+                    String username = prompter.prompt( usernameItem.getPromptText() );
+                    usernameItem.setValue( username );
+                }
+                else if ( item instanceof CredentialItem.InformationalMessage )
+                {
+                    prompter.showMessage( item.getPromptText() );
+                }
+                else
+                {
+                    throw new UnsupportedCredentialItem( uri, item.getClass().toString() );
+                }
+            }
+            catch ( PrompterException e )
+            {
+                throw new IllegalStateException( "Can not prompt user " + e.getMessage() , e );
+            }
+        }
+        return true;
+    }
+
+    public void setInteractive( boolean isInteractive )
```

Comment:
same here

Reply:
Done in https://github.com/apache/maven-scm/pull/154/commits/617c43e02f33474dc9fc2c17ee400f874d18d789